src/components/ItemDisplay.jsx

```diff
@@ -0,0 +1,66 @@
+import React, { useRef, useEffect } from 'react';
+import logo from '../assets/one.jpg';
+import { format } from 'date-fns';
+
+const ItemDisplay = ({ setShowModal, closeModal, details }) => {
+	const mainRef = useRef(null);
+
+	useEffect(() => {
+		mainRef.current = document.getElementById('modal');
+
+		window.addEventListener('click', function (e) {
+			if (e.target === mainRef.current) {
+				setShowModal(false);
+			}
+		});
+	}, [setShowModal]);
+
+	const convertFromNull =
```

Comment:
```suggestion
	const dateLastPurchased =
```

This reads better IMO

Reply:
Updated